src/main/groovy/com/google/gcloud/exec/io/WatchableInputStreamConsumer.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.gcloud.exec.io
+
+import org.codehaus.groovy.runtime.IOGroovyMethods
+import org.gradle.api.logging.LogLevel
+import org.gradle.api.logging.Logger
+
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+/**
+ * A stream consume and watcher that will consume and inspect only the input stream
+ * from a process, ensure the stderr redirects to stdin for this implementation
+ *
+ * Set the stream consumer on this object directly (StreamWatcher.setInspector) after
+ * instantiating the object.
+ */
+class WatchableInputStreamConsumer implements StreamConsumer, StreamWatcher {
+
+    final ExecutorService executorService
+    final Logger logger
+    final LogLevel logLevel
+
+    WatchableInputStreamConsumer(Logger logger, LogLevel logLevel) {
+        executorService = Executors.newSingleThreadExecutor()
+        this.logger = logger
+        this.logLevel = logLevel
+    }
+
+    /**
+     * Only consumes input stream, make sure you merge streams in the process builder
+     * before calling this
+     */
+    @Override
+    void consumeStreams(Process process) {
```

Comment:
So changing the interface turned out to be a worse idea.

Technically the stream consumer is capable of consuming all streams, and the runner is agnostic to the implementation. I think this is okay. Sending in the right processStarter and streamConsumer is up to the developer.